contrib/kube-prometheus/manifests/prometheus/prometheus-k8s-rules.yaml

```diff
@@ -516,4 +516,52 @@ data:
         summary = "Kubelet is close to pod limit",
         description = "Kubelet {{$labels.instance}} is running {{$value}} pods, close to the limit of 110",
       }
-    
+
+    ALERT K8SPodWaiting
+      IF sum(sum_over_time(kube_pod_container_status_waiting[5m]) > 0 ) BY (job, exported_pod, instance, namespace)
+      FOR 5m
+      LABELS {
+        service = "k8s",
+        severity = "critical",
+      }
+      ANNOTATIONS {
+        summary = "Pod is waiting too long",
+        description = "Pod {{ $labels.namespace }}/{{ $labels.exported_pod }} is in waiting state",
+      }
+
+    ALERT K8sPodRestartingTooMuch
+      IF sum (rate(kube_pod_container_status_restarts[5m]) > 0) BY(job, exported_pod, instance, namespace)
+      FOR 5m
+      LABELS {
+        service = "k8s",
+        severity = "critical",
+      }
+      ANNOTATIONS {
+        summary = "Pod is in a restart loop",
+        description = "Pod {{ $labels.namespace }}/{{ $labels.exported_pod }} is restarting constantly",
+      }
+
+  docker.rules: |+
+    ALERT DockerHasErrors
+      IF irate(kubelet_docker_operations_errors[10m])  > 0
+      FOR 1m
+      LABELS {
+        service = "docker",
+        severity = "critical",
```

Comment:
Yes, same here. Docker daemon erroring does not necessarily impact the service. In case Kubernetes is a service provided by ops to developers, there are likely higher-level metrics indicating if devs cannot deploy for example.